api/src/pcapi/routes/public/individual_offers/v1/bookings.py

```diff
@@ -194,6 +194,50 @@ def validate_booking_by_token(token: str) -> None:
     bookings_api.mark_as_used(booking)
 
 
+@blueprint.v1_bookings_blueprint.route("/keep/token/<token>", methods=["PATCH"])
+@spectree_serialize(
+    on_success_status=204,
+    api=blueprint.v1_bookings_schema,
+    tags=[constants.BOOKING_TAG],
+    resp=SpectreeResponse(
+        **(
+            BOOKINGS_API_BASE_CODE_DESCRIPTIONS
+            | {
+                "HTTP_204": (None, "The booking's validation has been cancelled successfully"),
+                "HTTP_403": (
+                    None,
+                    "This booking's validation cannot cancelled because you do not have the proper right or because the token's status is not used",
+                ),
+                "HTTP_410": (None, "This booking cannot be unvalidated"),
+            }
+        )
+    ),
+)
+@api_key_required
+def cancel_booking_validation_by_token(token: str) -> None:
+    """
+    Cancel a booking's validation.
+
+    To mark a booking as unused.
+    """
+    booking = _get_booking_by_token(token)
+    if booking is None:
+        raise api_errors.ApiErrors(errors={"global": "This countermark cannot be found"}, status_code=404)
+
+    try:
+        bookings_api.mark_as_unused(booking)
+    except exceptions.BookingIsAlreadyRefunded:
+        raise api_errors.ApiErrors(errors={"payment": "This booking has been reimbursed"}, status_code=403)
+    except exceptions.BookingIsNotUsed:
+        raise api_errors.ApiErrors(errors={"booking": "This booking has not been used"}, status_code=403)
+    except exceptions.BookingIsAlreadyCancelled:
+        raise api_errors.ApiErrors(errors={"booking": "This booking has been canceled"}, status_code=410)
```

Comment:
typo (tu as mis 2 "l" dans d'autres message, autant être homogène) : 

```suggestion
        raise api_errors.ApiErrors(errors={"booking": "This booking has been cancelled"}, status_code=410)
```